### PR TITLE
feat: Add -d/--dev common command-line flag to put service in Dev Mode

### LINKED ISSUE #509: Add dev command line flag to apply "Host" env overrides when running in hybrid mode
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for Config processing

### Description [**REQUIRED**]
 Currently with the new Common Config, the common `Host` settings, i.e MessageBus.Host, etc.,  are in common config in the Config Provider. When running from command line in hybrid mode with Config Provider, the common `Host` settings have been overridden with the docker network names and are not valid when running service outside docker. 

The current work around is to have these common `Host` settings commented out every services' local private configuration file. When running the service in hybrid mode with Config Provider, the developer must uncomment these settings.

Example:
```yaml
# uncomment when running from command-line in hybrid mode with -cp -o flags
#Clients:
#  core-metadata: # Used for version check on start-up comes from Common Config
#      Host: "localhost"
#Registry:
#  Host: "localhost"
#Database:
#  Host: "localhost"

MessageBus:
#  Host: localhost # uncomment when running from command-line in hybrid mode
```
### Describe the solution you'd like
 Add the new `--hybrid` command flag which will enable Env Overriding of these `Host` settings when pulled from the Configuration Provider. This will be limited to only environment variables that end in `_HOST` so that it is not a end around to re-enable Env Overrides as highest precedence as is in V2.

### Describe alternatives you've considered
 Use current work around of commented out settings which clutter the service config files

### PR DESCRIPTION
**Pull request by lenny-intel**
Dev Mode is used when running service locally with the other services running in Docker (aka hybrid mode). This causes all the `Host` settings pulled from common configuration to be overridden with the value `localhost`.

closes #509

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
 TBD

## Testing Instructions
Use in edgex-go, app sdk and device sdk and make appropriate code changes
Build and run core/support services with "-cp -d" options with other services running in docker
Verify service bootstraps properly
Build and run device-simple with "-cp -d" options with other services running in docker
Verify service bootstraps properly
Build and run app-template with "-cp -d" options with other services running in docker
Verify service bootstraps properly

### CODE REVIEW

File: config/types.go [hunk 1]
```diff
@@ -226,11 +226,12 @@ type InsecureSecretsInfo struct {
...
-	Service      ServiceInfo
+	Clients      map[string]*ClientInfo
+	Service      *ServiceInfo
 	Config       ConfigProviderInfo
-	Registry     RegistryInfo
-	MessageBus   MessageBusInfo
+	Registry     *RegistryInfo
+	MessageBus   *MessageBusInfo
+	Database     *Database
 	ExternalMQTT ExternalMQTTInfo
```
Comment (cloudxxx8) [inline config/types.go:235]: why don't we use pointer for all of the fields?

Comment (lenny-intel) [reply config/types.go:235]: Bootstrap code doesn't need to change them, so didn't use pointer, but can for consistency.

Comment (lenny-intel) [reply config/types.go]: Reworked

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**